servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/AbstractLifecycleObserverHttpFilter.java

```diff
@@ -288,8 +289,8 @@ private static void safeReport(final Consumer<Throwable> onError, final Throwabl
         try {
             onError.accept(t);
         } catch (Throwable unexpected) {
-            unexpected.addSuppressed(t);
-            LOGGER.warn("Unexpected exception from {} while reporting a '{}' event", observer, eventName, unexpected);
+            LOGGER.warn("Unexpected exception from {} while reporting a '{}' event",
+                    observer, eventName, addSuppressed(unexpected, t));
```

Comment:
I preferred the version that has the function with side-effects as a separate statement. It is too easy to miss it here as an argument to logger. Pure functions are of less concern.